include/gsl/gsl_assert

```diff
@@ -68,20 +68,54 @@
 
 namespace gsl
 {
+
 struct fail_fast : public std::runtime_error
 {
     explicit fail_fast(char const* const message) : std::runtime_error(message) {}
 };
-}
+namespace details
+{
+template <typename Exception>
+void throw_exception(Exception&& exception);
```

Comment:
This forward declaration - and the declaration on line 91 - should be `[[noreturn]]` as well to make it clear that users may not define a `throw_exception` that returns.